### PR TITLE
build-script: don't disable iOS device tests based on the host variable

### PR DESCRIPTION
**Pull request by gribozavr**
Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by gribozavr** [comment]
@swift-ci Please smoke test

File: utils/build-script [hunk 1]
```diff
@@ -445,18 +445,12 @@ class BuildScriptInvocation(object):
             self.platforms_to_skip_test.add(StdlibDeploymentTarget.Cygwin)
         if args.skip_test_osx:
             self.platforms_to_skip_test.add(StdlibDeploymentTarget.OSX)
-        if args.skip_test_ios_host:
-            self.platforms_to_skip_test.add(StdlibDeploymentTarget.iOS)
```
Comment (ddunbar) [inline utils/build-script]: Aren't we *supposed* to disable these based on `skip_test_ios`, though? ... etc for lines below ...

**Comment by ddunbar** [comment]
LGTM

File: utils/build-script (hunk 1 above)
Comment (gribozavr) [inline utils/build-script]: `skip_test_ios` is a shorthand for `skip_test_ios_host` and `skip_test_ios_simulator`.  Let's just disable them unconditionally.

**Pull request merged**